arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -625,157 +646,139 @@ public inline fun <A, B> List<A>.reduceRightNull(
 }
 
 /**
- * Returns a [List<Pair<A?, B?>>] containing the zipped values of the two lists with null for padding.
+ * Returns a [List] containing the zipped values of the two lists with null for padding.
  *
- * Example:
  * ```kotlin
  * import arrow.core.*
+ * import io.kotest.matchers.shouldBe
  *
- * //sampleStart
- * val padRight = listOf(1, 2).padZip(listOf("a"))        // Result: [Pair(1, "a"), Pair(2, null)]
- * val padLeft = listOf(1).padZip(listOf("a", "b"))       // Result: [Pair(1, "a"), Pair(null, "b")]
- * val noPadding = listOf(1, 2).padZip(listOf("a", "b"))  // Result: [Pair(1, "a"), Pair(2, "b")]
- * //sampleEnd
- *
- * fun main() {
- *   println("padRight = $padRight")
- *   println("padLeft = $padLeft")
- *   println("noPadding = $noPadding")
+ * fun test() {
+ *   listOf(1, 2).padZip(listOf("a")) shouldBe listOf(1 to "a", 2 to null)
+ *   listOf(1).padZip(listOf("a", "b")) shouldBe listOf(1 to "a", null to "b")
+ *   listOf(1).padZip(listOf("a", "b")) shouldBe listOf(1 to "a", null to "b")
  * }
  * ```
  * <!--- KNIT example-iterable-03.kt -->
+ * <!--- TEST lines.isEmpty() -->
  */
 public fun <A, B> Iterable<A>.padZip(other: Iterable<B>): List<Pair<A?, B?>> =
-  align(other) { ior ->
-    ior.fold(
-      { it to null },
-      { null to it },
-      { a, b -> a to b }
-    )
-  }
+  padZip(other) { a, b -> a to b }
 
 /**
- * Returns a [List<C>] containing the result of applying some transformation `(A?, B?) -> C`
- * on a zip.
+ * Returns a [List] containing the result of applying some transformation `(A?, B?) -> C` on a zip.
  *
- * Example:
  * ```kotlin
  * import arrow.core.*
+ * import io.kotest.matchers.shouldBe
  *
- * //sampleStart
- * val padZipRight = listOf(1, 2).padZip(listOf("a")) { l, r -> l to r }     // Result: [Pair(1, "a"), Pair(2, null)]
- * val padZipLeft = listOf(1).padZip(listOf("a", "b")) { l, r -> l to r }    // Result: [Pair(1, "a"), Pair(null, "b")]
- * val noPadding = listOf(1, 2).padZip(listOf("a", "b")) { l, r -> l to r }  // Result: [Pair(1, "a"), Pair(2, "b")]
- * //sampleEnd
- *
- * fun main() {
- *   println("padZipRight = $padZipRight")
- *   println("padZipLeft = $padZipLeft")
- *   println("noPadding = $noPadding")
+ * fun test() {
+ *   listOf(1, 2).padZip(listOf("a")) { l, r -> l to r } shouldBe listOf(1 to "a", 2 to null)
+ *   listOf(1).padZip(listOf("a", "b")) { l, r -> l to r } shouldBe listOf(1 to "a", null to "b")
+ *   listOf(1).padZip(listOf("a", "b")) { l, r -> l to r } shouldBe listOf(1 to "a", null to "b")
```

Comment:
Duplicated of the example above
```suggestion
 *   listOf(1, 2).padZip(listOf("a", "b")) { l, r -> l to r } shouldBe listOf(1 to "a", 2 to "b")
```

Reply:
I will require running `knit` command again